utils/lib/connection_database.sql

```diff
@@ -0,0 +1,247 @@
+--
+-- This table is the list of these direct connections.
+CREATE TABLE edge_with_mux(
+  pkey INTEGER PRIMARY KEY,
+  src_wire_pkey INT,
+  dest_wire_pkey INT,
+  pip_in_tile_pkey INT,
+  FOREIGN KEY(src_wire_pkey) REFERENCES wire(pkey),
+  FOREIGN KEY(dest_wire_pkey) REFERENCES wire(pkey),
+  FOREIGN KEY(pip_in_tile_pkey) REFERENCES pip_in_tile(pkey)
+);
+
+-- Table of graph nodes.  This is a direction mapping of an VPR rr_node
+-- instance.
+CREATE TABLE graph_node(
+  pkey INTEGER PRIMARY KEY,
+  graph_node_type INT,
+  track_pkey INT,
+  node_pkey INT,
+  x_low INT,
+  x_high INT,
+  y_low INT,
+  y_high INT,
+  ptc INT,
+  capacity INT,
+  FOREIGN KEY(track_pkey) REFERENCES track(pkey),
+  FOREIGN KEY(node_pkey) REFERENCES node(pkey)
+);
+
+-- Table of wires.  This table is the complete list of all wires in the
+-- grid. All wires will belong to exactly one node.
+--
+-- Rows will relate back to their parent tile, and generic wire instance.
+--
+-- If the wire is connected to both a site pin and a pip, then
+-- top_graph_node_pkey, bottom_graph_node_pkey, left_graph_node_pkey, and
+-- right_graph_node_pkey will be set to the IPIN or OPIN instances, based
+-- on the pin directions for the tile.
+--
+-- If the wire is a member of a routing node, then graph_node_pkey will be
+-- set to the graph_node this wire is a member of.
+CREATE TABLE wire(
+  pkey INTEGER PRIMARY KEY,
+  node_pkey INT,
+  tile_pkey INT,
+  wire_in_tile_pkey INT,
+  graph_node_pkey INT,
+  top_graph_node_pkey INT,
+  bottom_graph_node_pkey INT,
+  left_graph_node_pkey INT,
+  right_graph_node_pkey INT,
+  FOREIGN KEY(node_pkey) REFERENCES node(pkey),
+  FOREIGN KEY(tile_pkey) REFERENCES tile(pkey),
+  FOREIGN KEY(wire_in_tile_pkey) REFERENCES wire_in_grid(pkey)
+  FOREIGN KEY(graph_node_pkey) REFERENCES graph_node(pkey)
+  FOREIGN KEY(top_graph_node_pkey) REFERENCES graph_node(pkey)
+  FOREIGN KEY(bottom_graph_node_pkey) REFERENCES graph_node(pkey)
+  FOREIGN KEY(left_graph_node_pkey) REFERENCES graph_node(pkey)
+  FOREIGN KEY(right_graph_node_pkey) REFERENCES graph_node(pkey)
+);
+
+-- Table of switches.
+CREATE TABLE switch(
+  pkey INTEGER PRIMARY KEY,
+  name TEXT
+);
+
+-- Table of graph edges.
+CREATE TABLE graph_edge(
+  src_graph_node_pkey INT,
+  dest_graph_node_pkey INT,
+  switch_pkey INT,
+  track_pkey INT,
+  tile_pkey INT,
+  pip_in_tile_pkey INT,
+  FOREIGN KEY(src_graph_node_pkey) REFERENCES graph_node(pkey),
+  FOREIGN KEY(dest_graph_node_pkey) REFERENCES graph_node(pkey)
+  FOREIGN KEY(track_pkey) REFERENCES track(pkey)
+  FOREIGN KEY(tile_pkey) REFERENCES tile(pkey)
+  FOREIGN KEY(pip_in_tile_pkey) REFERENCES pip(pkey)
+);
+
+-- channel, x_list and y_list are direct mappings of the channel object
+-- present in the rr_graph.
+CREATE TABLE channel(
+  chan_width_max INT,
+  x_min INT,
+  y_min INT,
+  x_max INT,
+  y_max INT
+);
+CREATE TABLE x_list(
+    idx INT,
+    info INT
+);
+CREATE TABLE y_list(
+    idx INT,
+    info INT
+);
+
+INSERT INTO
```

Comment:
Leave the initial switch names in the python, this will become dynamic in the future.

Reply:
Done.